Backport of upstream rules_pkg changes (drop when bumping rules_pkg to a
release containing them):

pkg_deb: Add `installed_size` attribute
https://github.com/bazelbuild/rules_pkg/pull/1068

pkg_deb: add installed_size_file attribute
https://github.com/bazelbuild/rules_pkg/pull/1083

Adapted to the list based `args.append()` of rules_pkg 1.1.0.

--- a/pkg/private/deb/deb.bzl
+++ b/pkg/private/deb/deb.bzl
@@ -133,6 +133,13 @@
     elif ctx.attr.depends:
         args += ["--depends=" + d for d in ctx.attr.depends]
 
+    if ctx.attr.installed_size_file:
+        if ctx.attr.installed_size:
+            fail("Both installed_size and installed_size_file attributes were specified")
+        args.append("--installed_size=@" + ctx.file.installed_size_file.path)
+        files.append(ctx.file.installed_size_file)
+    elif ctx.attr.installed_size:
+        args.append("--installed_size=" + ctx.attr.installed_size)
     if ctx.attr.priority:
         args.append("--priority=" + ctx.attr.priority)
     if ctx.attr.section:
@@ -314,6 +321,17 @@
             See http://www.debian.org/doc/debian-policy/ch-archive.html#s-subsections.""",
         ),
         "homepage": attr.string(doc = """The homepage of the project."""),
+        "installed_size": attr.string(
+            doc = """The approximate total size (in KiB) of the package's installed files.
+            Corresponds to the Installed-Size field in the control file.
+            See https://www.debian.org/doc/debian-policy/ch-controlfields.html#installed-size.
+            Mutually exclusive with `installed_size_file`.""",
+        ),
+        "installed_size_file": attr.label(
+            doc = """File that contains the approximate total size (in KiB) of the
+            package's installed files. Mutually exclusive with `installed_size`.""",
+            allow_single_file = True,
+        ),
         "license": attr.string(doc = """The license of the project."""),
         "breaks": attr.string_list(
             doc = """See http://www.debian.org/doc/debian-policy/ch-relationships.html#s-binarydeps.""",
